subiquity/common/api/server.py

```diff
@@ -131,6 +134,18 @@ async def handler(request):
                     args['context'] = context
                 if 'request' in impl_params:
                     args['request'] = request
+                if not getattr(
+                        definition, 'allowed_before_start', False):
+                    # Most endpoints should not be responding to requests
+                    # before the controllers have started, that's just bound to
+                    # be a big pool of timing bugs that we want nothing to do
+                    # with.  A few chosen methods should be safe, so allow
+                    # those to opt-in.  Everybody else blocks until the
+                    # controllers complete their start().
+                    if not controller.app.controllers_have_started.is_set():
+                        log.debug(f'{request.path} waiting on start')
+                        await controller.app.controllers_have_started.wait()
+                        log.debug(f'resume {request.path}')
```

Comment:
I guess the logging is nice but "if not event.is_set(): await event.wait()" is kind of a strange pattern

Reply:
It's the sort of thing that I hope is vary rare to see in the logs, but if it does show up then it may be a clue as to other problems.